torch/nn/functional.py

```diff
@@ -2609,7 +2609,7 @@ def smooth_l1_loss(input, target, size_average=None, reduce=None, reduction='mea
     if size_average is not None or reduce is not None:
         reduction = _Reduction.legacy_get_string(size_average, reduce)
     if target.requires_grad:
-        ret = _smooth_l1_loss(input, target)
+        ret = _smooth_l1_loss(input, target, delta=delta)
```

Comment:
You should have also changed lines 2617 (and the corresponding C++ / CUDA implementations).

As it currently stands, the `delta` is not used if `target.requires_grad == False`.

This should have had a test that would have catched this